### PR TITLE
Use .NET 8 SDK

### PR DESCRIPTION
**Pull request by YuKitsune**
[sc-65325]

Depends on:
- [x] https://github.com/OctopusDeploy/OctopusClients/pull/807

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -120,7 +138,7 @@ jobs:
   test-linux-distributions:
     strategy:
       matrix:
-        os: [ubuntu16, ubuntu18, ubuntu20, debian-buster, fedora35, centos7, rhel9, rocky9, amazon2, opensuse-leap15, opensuse-tumbleweed ]
+        os: [ubuntu16, ubuntu18, ubuntu20, debian-buster, fedora35, rhel9, rocky9, amazon2, opensuse-leap15, opensuse-tumbleweed ]
```
Comment (YuKitsune) [inline .github/workflows/main.yaml:141]: [.NET8 isn't supported on Centos](https://learn.microsoft.com/en-us/dotnet/core/install/linux-centos#supported-distributions).

File: source/Octopus.Client.Tests/Octopus.Client.Tests.csproj [hunk 2]
```diff
@@ -12,18 +12,18 @@
     <LangVersion>8</LangVersion>
   </PropertyGroup>
   <PropertyGroup Condition="!$([MSBuild]::IsOSUnixLike())">
-    <TargetFrameworks>net462;net48;net6.0</TargetFrameworks>
+    <TargetFrameworks>net462;net48;net6.0;net8.0</TargetFrameworks>
   </PropertyGroup>
   <PropertyGroup Condition="$([MSBuild]::IsOSUnixLike())">
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFrameworks>net6.0;net8.0</TargetFrameworks>
   </PropertyGroup>
```
Comment (YuKitsune) [inline source/Octopus.Client.Tests/Octopus.Client.Tests.csproj:19]: @borland Do we still need to target .NET 6?

Comment (borland) [reply source/Octopus.Client.Tests/Octopus.Client.Tests.csproj:19]: Yeah, if the library targets net6 then the tests should too

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -38,11 +38,13 @@ jobs:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0 # all
-      - name: Setup .NET 6.0
+
```
Comment (borland) [inline .github/workflows/main.yaml:41]: Because the tests target both 6 and 8, we should have both a setup net6 and a setup net8 action here

**Review by borland — APPROVED** [review]

**Pull request merged**